data_relay.py

```diff
@@ -51,82 +47,158 @@
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding'),
+            ('x', 'one byte of padding')
             ),
-        LOW_FREQ: ( # 75 bytes + 13 padding bytes
-            ('f', 'roll_ki'),
-            ('f', 'pitch_ki'),
-            ('f', 'yaw_ki'),
-            ('f', 'heading_kd'),
+        PACKET_TYPE_GAINS: ( # 80 bytes + 0 bytes of padding
+            ('f', 'roll_rate_kp'),
+            ('f', 'roll_rate_kd'),
+            ('f', 'roll_rate_ki'),
+            ('f', 'pitch_rate_kp'),
+            ('f', 'pitch_rate_kd'),
+            ('f', 'pitch_rate_ki'),
+            ('f', 'yaw_rate_kp'),
+            ('f', 'yaw_rate_kd'),
+            ('f', 'yaw_rate_ki'),
+            ('f', 'roll_angle_kp'),
+            ('f', 'roll_angle_kd'),
+            ('f', 'roll_angle_ki'),
+            ('f', 'pitch_angle_kp'),
+            ('f', 'pitch_angle_kd'),
+            ('f', 'pitch_angle_ki'),
             ('f', 'heading_kp'),
-            ('f', 'heading_ki'),
-            ('f', 'altitude_kd'),
             ('f', 'altitude_kp'),
-            ('f', 'altitude_ki'),
-            ('f', 'throttle_kd'),
-            ('f', 'throttle_kp'),
-            ('f', 'throttle_ki'),
-            ('f', 'flap_kd'),
-            ('f', 'flap_kp'),
-            ('f', 'flap_ki'),
-            ('f', 'path_gain'),
-            ('f', 'orbit_gain'),
-            ('h', 'autonomousLevel'),
-            ('h', 'startup_error_codes'),
-            ('h', 'startupSettings'),
-            ('h', 'dl_transmission_errors'),
-            ('h', 'ul_receive_errors'),
-            ('B', 'ul_rssi'),
-            ('B', 'uhf_rssi'),
-            ('B', 'uhf_link_quality'),
+            ('f', 'ground_speed_kp'),
+            ('f', 'path_kp'),
+            ('f', 'orbit_kp')
```

Comment:
I like the grouping of gains together, but you will notice it makes up the longest packet. This is a trade-off you may be okay with.

I would put RSSI values (or some of the error codes) in a different packet, where they are updated more often, as they are critical to the safety of the plane.